app/components/post_draft/index.ts

```diff
@@ -43,10 +43,7 @@ const enhanced = withObservables([], (ownProps: WithDatabaseArgs & OwnProps) =>
     );
 
     const canPost = combineLatest([channel, currentUser]).pipe(switchMap(([c, u]) => from$(hasPermissionForChannel(c, u, Permissions.CREATE_POST, false))));
-    let channelIsArchived = of$(ownProps.channelIsArchived);
-    if (!channelIsArchived) {
-        channelIsArchived = channel.pipe(switchMap((c) => of$(c.deleteAt !== 0)));
-    }
+    const channelIsArchived = channel.pipe(switchMap((c) => (ownProps.channelIsArchived ? of$(true) : of$(c.deleteAt !== 0))));
```

Comment:
Could this become a `@lazy isArchived = ...` property on the model? Might come in handy for others down the track - but non-urgent/blocking.

Reply:
I don't see how this can be a lazy query on the model really.
1. Uses a prop of the component
3. Access directly the delete_at property of the model